typify-impl/src/lib.rs

```diff
@@ -70,6 +76,7 @@ pub struct Type<'a> {
     type_entry: &'a TypeEntry,
 }
 
+#[allow(missing_docs)]
```

Comment:
I think we can document this; see inspiration below. Let me know if you'd like me to add suggestions.

Reply:
I've added the ones you've given, and added a few more.
My intention was to put `#[allow(missing_docs)]` where other people can replace it with real docs.

The goal of the PR isnt to document everything, but instead to ensure new enhancements must include docs at the outset.